### PR TITLE
Add USART support for LPC845

### PR DESCRIPTION
**Pull request by hannobraun**
@david-sawatzke Here's my work-in-progress. I'll try to finish this up and get it merged tomorrow.

Looks like you have some stuff in your PR that I'm missing. I'm going to integrate that after RustFest.

### CODE REVIEW

File: src/usart.rs [hunk 1]
```diff
@@ -452,41 +462,53 @@ impl Peripheral for pac::USART2 {
...
-            brgval  : brgval,
+            _clock: clock,
+            brgval: brgval,
         }
     }
 }
 
 
+/// Implemented for USART clock sources
+pub trait ClockSource {
```
Comment (david-sawatzke) [inline src/usart.rs]: The clock source stuff is the same for i2c & spi as well, so it probably makes sense to put it in syscon

File: src/syscon/frg.rs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            ///
+            /// This is an internal implementation detail.
+            pub struct $mult;
+
+            /// Represents and instance of FRGCLKSEL
+            ///
+            /// This is an internal implementation detail.
+            pub struct $clksel;
+
+            reg_cluster!($div,    FRGDIV,    pac::SYSCON, frg0, frgdiv);
```
Comment (david-sawatzke) [inline src/syscon/frg.rs]: This only works for `frg0`

**Comment by david-sawatzke** [comment]
Seems good enough as a POC otherwise

File: src/usart.rs (hunk 1 above)
Comment (hannobraun) [reply src/usart.rs]: Good point. I didn't want to make to too generic at this point, as my goal was to get something working quickly.

File: src/syscon/frg.rs (hunk 2 above)
Comment (hannobraun) [reply src/syscon/frg.rs]: Thanks, fixed.

**Comment by hannobraun** [comment]
Unfortunately I wasn't able to get this to work. Not sure why, and unfortunately I didn't bring a logic analyzer with me to Barcelona.

I'll leave this open. I'll try to find some time to merge our two approaches later.

**Comment by hannobraun** [comment]
Update: I don't think I'll be able to finish this any time soon. If anyone else is interested, feel free to take it.

**Comment by hannobraun** [comment]
Closing in favor of #175. It's more fully featured, and has the distinction of actually working.

**Pull request closed without merging**